scipy/stats/stats.py

```diff
@@ -2312,6 +2318,275 @@ def zmap(scores, compare, axis=0, ddof=0):
         return (scores - mns) / sstd
 
 
+def _iqr_scale_to_num(scale):
+    """
+    A helper function to convert the 'scale' argument of `iqr` to a
+    number.
+
+    scale: str
+        If scale is one of ['raw', 'normal'], this function will map the
+        string to a value.
+
+    The concept for this function is borrowed heavily from the
+    `numpy.histogram` helper function `_hist_optim_numbins_estimator`.
+    """
+    # See https://en.wikipedia.org/wiki/Robust_measures_of_scale
+    scale_conversions = {'raw': 1.0,
+                         'normal': special.erfinv(0.5) * 2.0 * math.sqrt(2.0)}
+    try:
+        return scale_conversions[scale.lower()]
+    except KeyError:
+        raise ValueError("{0} not a valid scale for `iqr`".format(scale))
```

Comment:
I don't like the use of `try/except` instead of `if key not in x` checks is because on Python3 this will have an extra-long traceback, where it says an error was thrown during the handling of another error.

It also seems a little overkill to have this in a separate function given how simple it is (and it's only used in one place).